### PR TITLE
Import from ICS files

### PR DESCRIPTION
**Pull request by fabiozaramella**
This PR add the option to import calendars from ICS files from the gear menu, fixing #275 and fixing #479.

It also resolves some incorrect displaying of the number of calendars that are going to be imported.

### CODE REVIEW

**Comment by tintou** [comment]
FYI, [we should stop using Gtk.Dialog(Native).run](https://developer.gnome.org/gtk4/unstable/ch41s02.html#id-1.7.4.4.75) as it is removed from Gtk4, we should use dialog.response.connect () with dialog.show()

**Comment by fabiozaramella** [comment]
> FYI, [we should stop using Gtk.Dialog(Native).run](https://developer.gnome.org/gtk4/unstable/ch41s02.html#id-1.7.4.4.75) as it is removed from Gtk4, we should use dialog.response.connect () with dialog.show()

Thank you for this! I'll update the PR then. 👍🏻

File: src/SourceDialog/SourceSelector.vala [hunk 1]
```diff
@@ -71,6 +75,43 @@ public class Maya.View.SourceSelector : Gtk.Popover {
             create_source ();
             return Gdk.EVENT_STOP;
         });
+
+        import_calendar_button.button_release_event.connect (() => {
+            GLib.File[] files = null;
```
Comment (tintou) [inline src/SourceDialog/SourceSelector.vala:80]: That can go in the `response` block

File: src/SourceDialog/SourceSelector.vala [hunk 2]
```diff
@@ -71,6 +75,43 @@ public class Maya.View.SourceSelector : Gtk.Popover {
...
+            file_chooser.set_local_only (true);
+            file_chooser.set_select_multiple (true);
+            file_chooser.set_filter (ics_filter);
+
+            file_chooser.show ();
+            this.hide ();
+
+            file_chooser.response.connect ((response_id) => {
+                if (response_id == Gtk.ResponseType.ACCEPT) {
+                    foreach (GLib.File selected_file in file_chooser.get_files ()) {
```
Comment (tintou) [inline src/SourceDialog/SourceSelector.vala:102]:
```suggestion
                    foreach (unowned GLib.File selected_file in file_chooser.get_files ()) {
```

File: src/ImportDialog.vala [hunk 3]
```diff
@@ -41,13 +41,13 @@ public class Maya.View.ImportDialog : Granite.MessageDialog {
                 critical (e.message);
             }
 
-            secondary_text = _("Events from \"%s\" will be merged with this calendar:").printf (name);
+            secondary_text = _(("Events from \"%s\" will be merged with this calendar:").printf (name));
```
Comment (tintou) [inline src/ImportDialog.vala:44]: This is wrong, the text has to be translated first then replaced

Comment (fabiozaramella) [reply src/ImportDialog.vala]: You're right, sorry. I've found the problem, someone mistranslated it in Italian using %d instead of %s... 🤦🏻‍♂️️

**Review by danrabbit — CHANGES_REQUESTED** [review]
I guess this exposes that we're using pretty large row spacing here. Probably what we should do is remove row spacing and then add 3px top and bottom margin to the separator. That would look more natural for a menu

File: src/SourceDialog/SourceSelector.vala [hunk 4]
```diff
@@ -51,13 +51,17 @@ public class Maya.View.SourceSelector : Gtk.Popover {
         var add_calendar_button = new Gtk.ModelButton ();
         add_calendar_button.text = _("Add New Calendar…");
 
+        var import_calendar_button = new Gtk.ModelButton ();
+        import_calendar_button.text = _("Import From ICS File…");
```
Comment (danrabbit) [inline src/SourceDialog/SourceSelector.vala:55]: I guess we should probably use the user-facing name of the file format and not the extension

```suggestion
        var import_calendar_button = new Gtk.ModelButton () {
            text = _("Import iCalendar File…")
        };
```

Comment (fabiozaramella) [reply src/SourceDialog/SourceSelector.vala:55]: Makes sense 👍🏼️

**Review by tintou — APPROVED** [review]

**Comment by fabiozaramella** [comment]
> I guess this exposes that we're using pretty large row spacing here. Probably what we should do is remove row spacing and then add 3px top and bottom margin to the separator. That would look more natural for a menu

Is it better now? It looks ok to me with these margins.

**Review by danrabbit — APPROVED** [review]
Looks great, nice work!

**Pull request merged**